Mainnet/StandardToken/StandardToken/StandardToken.cs

```diff
@@ -13,7 +14,7 @@ public class StandardToken : SmartContract, IStandardToken
     /// <param name="totalSupply">The total token supply.</param>
     /// <param name="name">The name of the token.</param>
     /// <param name="symbol">The symbol used to identify the token.</param>
-    public StandardToken(ISmartContractState smartContractState, ulong totalSupply, string name, string symbol)
+    public StandardToken(ISmartContractState smartContractState, Amount totalSupply, string name, string symbol)
```

Comment:
Can `Amount` be changed to `UInt256`? How would `UInt128` be used? If possible it would be nice to use those types explicitly as `UInt256` or `UInt128`.

Also, while we're here, can we add the Decimals property and methods from this PR https://github.com/stratisproject/CirrusSmartContracts/pull/17

Reply:
I think it is better to use UInt256 explicitly in here otherwise it is confusing.

Reply:
Btw it would be better to update testnet files first then we can create mainnet after being sure it is fine. Currently these are test files.

Reply:
Now decimals update pr duplicated :) I want to push these changes first then can you apply your changes ?

Reply:
@quantumagi add decimals parameter to constructor and set it to property. Decimals methods should to be updated as Property getters/setters. @quantumagi

Reply:
@YakupIpek @quantumagi - https://github.com/stratisproject/CirrusSmartContracts/pull/17 is ready for review and merging, it's probably easiest/best to review and merge that into master, then update this branch with those changes and update the README included in that PR with the latest bytecode and hash.